src/resolve/resolved-dns-query.c

```diff
@@ -587,8 +586,7 @@ void dns_query_complete(DnsQuery *q, DnsTransactionState state) {
         q->state = state;
 
         if (state == DNS_TRANSACTION_SUCCESS && set_size(q->manager->varlink_subscription) > 0) {
-                DnsQuestion *question = q->request_packet ? q->request_packet->question : NULL;
-                const char *query_name = question ? dns_question_first_name(question) : q->request_name;
+                const char *query_name = dns_question_first_name(q->request_packet ? q->request_packet->question : q->question_utf8);
```

Comment:
Check on centos (but not on ubuntu) is now failing and this is the only functional change, so I assume it's due to this.

Test is receiving `{"parameters":{"addresses":[{"ifindex":6,"family":2,"address":[10,0,0,14],"type":"A"}],"name":"follow10.so.close.signed.test"},"continues":true}` but it's expecting `cname-chain.signed.test` as the name. @mrc0mmand Any idea? This was changed following @poettering request

Reply:
No idea, unfortunately. I've also noticed there's a race condition somewhere where we might miss some notifications, as already happened in several unrelated PRs, e.g.: https://jenkins-systemd.apps.ocp.ci.centos.org/job/upstream-centos8/11030/artifact//systemd-centos-ci/artifacts_wf1eehdn/testsuite-logs-upstream.bBa/TEST-75-RESOLVED_FAIL.log

Reply:
hmm, so we follow a cname chain on that request. Question is what we should return to clients here. Pretty sure we should return one varlink message for each element in the chain.

Reply:
so, basically, don't bother qith `q->request_packet`, it's never what you want. Aloway use `q->question_utf8`